src/RestApi/StoreApi/Utilities/CartController.php

```diff
@@ -364,25 +364,29 @@ public function get_cart_coupons( $callback = null ) {
 	}
 
 	/**
-	 * Get shipping packages from the cart and format as required.
+	 * Get shipping packages from the cart with calculated shipping rates.
 	 *
-	 * @param bool  $calculate_rates Should rates for the packages also be returned.
-	 * @param array $destination Pass an address to override the package destination before calculation.
+	 * @todo this can be refactored once https://github.com/woocommerce/woocommerce/pull/26101 lands.
+	 *
+	 * @param bool $calculate_rates Should rates for the packages also be returned.
 	 * @return array
 	 */
-	public function get_shipping_packages( $calculate_rates = true, $destination = [] ) {
+	public function get_shipping_packages( $calculate_rates = true ) {
 		$cart     = $this->get_cart_instance();
 		$packages = $cart->get_shipping_packages();
 
 		// Add package ID to array.
 		foreach ( $packages as $key => $package ) {
-			$packages[ $key ]['package_id'] = $key;
-
-			if ( ! empty( $destination ) ) {
-				$packages[ $key ]['destination'] = $destination;
+			if ( ! isset( $packages[ $key ]['package_id'] ) ) {
+				$packages[ $key ]['package_id'] = $key;
 			}
 		}
 
+		// See if we need to calculate anything.
+		if ( ! WC()->cart->needs_shipping() ) {
+			return [];
+		}
+
```

Comment:
Could this if clause be moved to the top of the function so we return early if it doesn't need shipping?